src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.cs

```diff
@@ -151,5 +154,37 @@ static CodeAction GetOptionalContructorParametersCodeAction(Document document, C
                     document, constructorCandidate, containingType, missingOptionalParameters, useSubMenuName);
             }
         }
+
+        public async Task<ImmutableArray<IntentProcessorResult>> ComputeIntentAsync(Document priorDocument, TextSpan priorSelection, Document currentDocument, string serializedIntentData, CancellationToken cancellationToken)
+        {
+            var actions = await AddConstructorParametersFromMembersAsync(priorDocument, priorSelection, cancellationToken).ConfigureAwait(false);
+            if (actions.IsEmpty)
+            {
+                return ImmutableArray<IntentProcessorResult>.Empty;
+            }
+
+            // This provider generates nested actions, so we need to flatten them before we compute the edits.
+            actions = actions.SelectMany(a => GetNestedActions(a)).ToImmutableArray();
+
+            using var _ = ArrayBuilder<IntentProcessorResult>.GetInstance(out var results);
+
+            foreach (var action in actions)
+            {
+                var addParamAction = (AddConstructorParametersCodeAction)action;
+
+                // This provider only returns CodeActions with a single ApplyChangesOperation
+                var operations = await addParamAction.GetOperationsAsync(cancellationToken).ConfigureAwait(false);
+                var applyChangesOperation = operations.Single() as ApplyChangesOperation;
+                Contract.ThrowIfNull(applyChangesOperation);
+
+                var intent = new IntentProcessorResult(applyChangesOperation.ChangedSolution, addParamAction.Title, addParamAction.ActionName);
+                results.Add(intent);
```

Comment:
seems strange that we return teh changed solution.  given that intents can only cahnge teh doc we're in, i would expect to pass in more like the new document.

Reply:
So in the long run we want to support more than one doc I think, so when I wrote the API shape I made it a solution.  The current handler does take that solution and only return the document matching the request though.

Reply:
👍